answer.txt

```diff
@@ -1,4 +1,43 @@
 • What is a gitignore file and what should it include for a Java project?
+- Gitignore file are those files which are not pushed in our remote repositories. For eg: unwanted jar files,
+critical secret keys are put in gitignore file.
+
+For a Java project, following should be included:
+
+# Compiled class file
+*.class
+
+# Log file
+*.log
+
+# BlueJ files
+*.ctxt
+
+# Mobile Tools for Java (J2ME)
+.mtj.tmp/
+
+# Package Files #
+*.jar
+*.war
+*.nar
+*.ear
+*.zip
+*.tar.gz
+*.rar
+
+# virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
+hs_err_pid*
+
+
 • What does fast-forward in git mean?
-• Why should youhavedifferent branches?
```

Comment:
Please elaborate on this answer

Reply:
There might be more than one person working on a single project. So branches make it easier.